pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -226,12 +226,13 @@ func (m *kubeGenericRuntimeManager) startContainer(podSandboxID string, podSandb
 		}
 		msg, handlerErr := m.runner.Run(kubeContainerID, pod, container, container.Lifecycle.PostStart)
 		if handlerErr != nil {
-			m.recordContainerEvent(pod, container, kubeContainerID.ID, v1.EventTypeWarning, events.FailedPostStartHook, msg)
+			s, _ := grpcstatus.FromError(handlerErr)
+			m.recordContainerEvent(pod, container, kubeContainerID.ID, v1.EventTypeWarning, events.FailedPostStartHook, s.Message())
```

Comment:
``handlerErr`` may be not a grpc error. 
``msg`` has wrapped the handlerErr on the func ``m.runner.Run``

Reply:
@dabaooline if so, use msg directly.